### ISSUE TITLE
Unminify JavaScript in the SuiteCRM repository

### ISSUE THREAD

**Issue by connorshea** [salesagility/SuiteCRM#7333]
Earlier today I opened #7332 to fix some minified JavaScript that was in the FP_Events module, and that made me curious to see if there were any other files that were commited to the git repo with minification. It looks like there are quite a few.

A lot are in the `include/` directory, and I won't include them in this list just because them being minified _sort of_ makes sense (we should really still fix it, though).

For example
- [ ] Most of the JS files in `install/`, e.g. `install/license.js`, `install/oc_install.js`, `install/site_config.js` etc. Though there are corresponding source files in `jssource/src_files/install` it looks like? So maybe that's fine.
- [ ] [`modules/calendar/Cal.js`](https://github.com/salesagility/SuiteCRM/blob/develop/modules/calendar/Cal.js)
- [ ] [`modules/ACLRoles/ACLRoles.js`](https://github.com/salesagility/SuiteCRM/blob/develop/modules/ACLRoles/ACLRoles.js)
- [ ] [`modules/AM_ProjectTemplates/jsclass_scheduler.js`](https://github.com/salesagility/SuiteCRM/blob/develop/modules/AM_ProjectTemplates/jsclass_scheduler.js)
- [ ] [`modules/Campaigns/DetailView.js`](https://github.com/salesagility/SuiteCRM/blob/develop/modules/Campaigns/DetailView.js)
- [ ] [`modules/Campaigns/WebToLead.js`](https://github.com/salesagility/SuiteCRM/blob/develop/modules/Campaigns/WebToLead.js)
- [ ] [`modules/Campaigns/wizard.js`](https://github.com/salesagility/SuiteCRM/blob/develop/modules/Campaigns/wizard.js)
- [ ] [`modules/Connectors/Connector.js`](https://github.com/salesagility/SuiteCRM/blob/develop/modules/Connectors/Connector.js)
- [ ] [`modules/Contacts/Contact.js`](https://github.com/salesagility/SuiteCRM/blob/develop/modules/Contacts/Contact.js)
- [ ] [`modules/Documents/documents.js`](https://github.com/salesagility/SuiteCRM/blob/develop/modules/Documents/documents.js)
- [ ] [`modules/EAPM/EAPMEdit.js`](https://github.com/salesagility/SuiteCRM/blob/develop/modules/EAPM/EAPMEdit.js)
- [ ] [`modules/EmailTemplates/EmailTemplate.js`](https://github.com/salesagility/SuiteCRM/blob/develop/modules/EmailTemplates/EmailTemplate.js)
- [ ] [`modules/FP_events/duration_dependency.js`](https://github.com/salesagility/SuiteCRM/blob/develop/modules/FP_events/duration_dependency.js`) (#7332)
- [ ] [`modules/FP_events/jsclass_scheduler.js`](https://github.com/salesagility/SuiteCRM/blob/develop/modules/FP_events/jsclass_scheduler.js`) (#7332)
- [ ] [`modules/Meetings/duration_dependency.js`](https://github.com/salesagility/SuiteCRM/blob/develop/modules/Meetings/duration_dependency.js)
- [ ] [`modules/Meetings/jsclass_scheduler.js`](https://github.com/salesagility/SuiteCRM/blob/develop/modules/Meetings/jsclass_scheduler.js)
- [ ] [`modules/MergeRecords/Merge.js`](https://github.com/salesagility/SuiteCRM/blob/develop/modules/MergeRecords/Merge.js)
- [ ] [`modules/Leads/Lead.js`](https://github.com/salesagility/SuiteCRM/blob/develop/modules/Leads/Lead.js)
- [ ] [`modules/InboundEmail/InboundEmail.js`](https://github.com/salesagility/SuiteCRM/blob/develop/modules/InboundEmail/InboundEmail.js)
- [ ] [`modules/Home/tour.js`](https://github.com/salesagility/SuiteCRM/blob/develop/modules/Home/tour.js)
- [ ] [`modules/Project/jsclass_scheduler.js`](https://github.com/salesagility/SuiteCRM/blob/develop/modules/Project/jsclass_scheduler.js)
- [ ] [`modules/Schedulers/Schedulers.js`](https://github.com/salesagility/SuiteCRM/blob/develop/modules/Schedulers/Schedulers.js)
- [ ] [`modules/Studio/studio.js`](https://github.com/salesagility/SuiteCRM/blob/develop/modules/Studio/studio.js)
- [ ] [`modules/Studio/studiodd.js`](https://github.com/salesagility/SuiteCRM/blob/develop/modules/Studio/studiodd.js)
- [ ] [`modules/Studio/studiotabgroups.js`](https://github.com/salesagility/SuiteCRM/blob/develop/modules/Studio/studiotabgroups.js)
- [ ] [`modules/Studio/ygDDListStudio.js`](https://github.com/salesagility/SuiteCRM/blob/develop/modules/Studio/ygDDListStudio.js)
- [ ] [`modules/UpgradeWizard/upgradeWizard.js`](https://github.com/salesagility/SuiteCRM/blob/develop/modules/UpgradeWizard/upgradeWizard.js)
- [ ] [`modules/Users/DetailView.js`](https://github.com/salesagility/SuiteCRM/blob/develop/modules/Users/DetailView.js)
- [ ] [`modules/Users/login.js`](https://github.com/salesagility/SuiteCRM/blob/develop/modules/Users/login.js)
- [ ] [`modules/Users/PasswordRequirementBox.js`](https://github.com/salesagility/SuiteCRM/blob/develop/modules/Users/PasswordRequirementBox.js)
- [ ] [`modules/Users/UserEditView.js`](https://github.com/salesagility/SuiteCRM/blob/develop/modules/Users/UserEditView.js)
- [ ] [`modules/Users/User.js`](https://github.com/salesagility/SuiteCRM/blob/develop/modules/Users/User.js)
- [ ] [`service/utils/SugarRest.js`](https://github.com/salesagility/SuiteCRM/blob/develop/service/utils/SugarRest.js)

Note: I found these by searching for `var` in VS Code and then looking through the results for lines that looked like `var this;var that;var them;var spaghetti;`. This list probably isn't comprehensive.

Most (all?) of these are from the 'Initial Commit' to the repo 6 years ago. I'm not sure if the original SugarCRM files exist somewhere that we can use, or if they were minified in the Sugar repo as well. Assuming we can't recover them from a copy of the Sugar CE repo, the best solution would probably be to do what I did in #7332 and just reformat the files using Prettier.

**Comment by Dillon-Brown** [comment]
Agreed, ideally we shouldn't be committing generated source files. I think we'd need to add rebuilding the minified JS to the installer or have something in the command line so you can pull down and install from the repo.

**Comment by connorshea** [comment]
Interestingly, most of the files have counterparts in the `jssource` directory, but not all of them. I formatted most of the files in my pull request, but I'm not sure if it makes sense to even merge the PR given that the source files are already in the repo.

@Dillon-Brown would changing the installation system to minify the JS be particularly difficult? That'd allow us to just remove most of this code.

**Comment by Dillon-Brown** [comment]
@connorshea I imagine it shouldn't be too difficult. We already have all the functionality to do this already, just need to be called during installation. It looks like there is around 630 files to remove here :O

**Comment by connorshea** [comment]
O.O
